runtime/appruntime/api/handler.go

```diff
@@ -233,6 +236,15 @@ func (d *Desc[Req, Resp]) executeEndpoint(c execContext, invokeHandler func(midd
 	var nextFn middleware.Next
 	numMiddleware := len(d.middleware)
 	nextFn = func(req middleware.Request) (resp middleware.Response) {
+		// Ensure the HTTP status code is correctly set in the response
+		defer func() {
+			// If no explicit HTTP status has been set, then we use the default for the type of error
+			// or if Err is nil, we'll set 200
+			if resp.HTTPStatus == 0 {
```

Comment:
One challenge with this is: What if a middleware returns a non-nil `Err` but keeps `HTTPStatus` as a `2xx` or `3xx` code? That would be quite confusing.

Reply:
I think this is fine;

- `return &MiddlewareResp { Err: err }` would set the status to 5xx or 4xx
- `return &MiddlewareResp { Err: err, Status: 201 }` would return 2xx

So maybe we just have an explicit `|| (resp.Err != nil && resp.Status == 200)` as the override?